apollo-router/src/request.rs

```diff
@@ -165,27 +165,84 @@ impl Request {
     ///
     /// An error will be produced in the event that the query string parameters
     /// cannot be turned into a valid GraphQL `Request`.
-    pub fn from_urlencoded_query(url_encoded_query: String) -> Result<Request, serde_json::Error> {
-        let urldecoded: serde_json::Value =
-            serde_urlencoded::from_bytes(url_encoded_query.as_bytes())
-                .map_err(serde_json::Error::custom)?;
+    pub(crate) fn batch_from_urlencoded_query(
+        url_encoded_query: String,
+    ) -> Result<Vec<Request>, serde_json::Error> {
+        let value: serde_json::Value = serde_urlencoded::from_bytes(url_encoded_query.as_bytes())
+            .map_err(serde_json::Error::custom)?;
 
-        let operation_name = if let Some(serde_json::Value::String(operation_name)) =
-            urldecoded.get("operationName")
-        {
-            Some(operation_name.clone())
+        Request::process_query_values(&value)
+    }
+
+    /// Convert Bytes into a GraphQL [`Request`].
+    ///
+    /// An error will be produced in the event that the query string parameters
+    /// cannot be turned into a valid GraphQL `Request`.
+    pub(crate) fn batch_from_bytes(bytes: &[u8]) -> Result<Vec<Request>, serde_json::Error> {
+        let value: serde_json::Value =
+            serde_json::from_slice(bytes).map_err(serde_json::Error::custom)?;
+
+        Request::process_batch_values(&value)
+    }
+
+    fn process_batch_values(value: &serde_json::Value) -> Result<Vec<Request>, serde_json::Error> {
+        let mut result = vec![];
```

Comment:
I think there is a way to predict the capacity needed and initialize it using `with_capacity()`

Reply:
done: a8359234e9910bb848692ab13d96a24f5ef07c0f